### DISCUSSION TITLE
Make lazygit floating window full screen?

### DISCUSSION THREAD

**Discussion by briandipalma** [LazyVim/LazyVim#219] [General]
I used to have my lazygit floating window fullscreen (take over all the space Neovim has available to it). Given it's a stand alone application and there isn't any use in seeing a bit of Neovim at the borders could I raise a PR to change it to be like that in LazyVim?

It makes it easier to read side by side diffs on smaller screens.

Something like this: https://github.com/briandipalma/ansible/blob/082f9d6ae7f97eb8a775b3a3b85e8a40007f37e6/roles/common/files/nvim/after/plugin/toggleterm.lua#L14 ?

**Comment by folke** [comment]
I personally prefer the current way.

Users can easily customize this like below:

```lua
local Util = require("lazyvim.util")

vim.keymap.set("n", "<leader>gg", function()
  Util.float_term({ "lazygit" }, {
    size = {
      width = 1,
      height = 1,
    },
  })
end, { desc = "Lazygit (cwd)" })
```

**Reply by folke** [reply]
I could provide some options to change the defaults for float_term though. Maybe that would be convenient.

**Reply by briandipalma** [reply]
Honestly it's not a major issue, I can copy the code you provided. It's just that even on a 2K/ res 27" screen I noticed the reduced code real estate on side by side diffs. They aren't even 80 columns each I think so there's a lot of wrapping. On a low res screen it'd be unusable.

**Reply by dpetka2001** [reply]
Yep I also use the above code to open `lazygit` in fullscreen mode. It's much more convenient for my laptop. But it's trivial to change a keymap according to your preferences, that I don't think we should change the defaults. Again, the defaults might not be to everyone's tastes but you can change them pretty easily most of the times. Other times you just need to ask for some help :smiley:

**Reply by asfaltboy** [reply]
I had to add the `esc_esc = false` option  to allow `esc` to go out of prompts. But, perhaps it's simpler to call the util plugin directly?:

```lua
local Lazygit = require("lazyvim.util.lazygit")
map("n", "<leader>gg", function()
  Lazygit.open({
    size = {
      width = 1,
      height = 1,
    },
  })
end, { desc = "Lazygit (cwd)" })
```

Or is this a bad idea?

**Reply by dpetka2001** [reply]
Not a bad idea at all. It's just according to one's preferences. I do a similar thing [here](https://github.com/dpetka2001/dotfiles/blob/66943e9c3e12b04d25a81a5719bca09cee0f826d/dot_config/nvim/lua/config/keymaps.lua#L34-L47).

**Reply by asfaltboy** [reply]
Your comment [here](https://github.com/dpetka2001/dotfiles/blob/66943e9c3e12b04d25a81a5719bca09cee0f826d/dot_config/nvim/lua/config/keymaps.lua#L34-L37) is interesting:

```
-- Don't use `esc_esc = false` to not conflict with Lazygit <Esc> and be able
-- to escape terminal mode and change tabs                                   
```

It would be cool to escape lazygit terminal floating window without closing it... but how do you close prompts? Did you map it to another key binding in lazygit config or something?

**Reply by dpetka2001** [reply]
Sorry but this was from old config days from when I was trying out lazygit, but since then I'm mostly doing my work on terminal command line with `git` commands. It might be that I overlooked some feature I wasn't using back then and wasn't aware if that change would involve some conflict in other lazygit's features. I was mostly using lazygit in a new Neovim tab and just wanted to escape the lazygit window and be able to switch between tabs, that's why I did that. I just mentioned it as an example that it's not a bad idea over to somebody's own preferences.

PS: But I guess the easiest way with the above setup `esc_esc = false` would be to just remap `return: "<c-c>"`.

**Comment by lkhphuc** [comment]
If you are used to working with vim's tab (i.e Diffview.nvim plugin), you can just create a mapping to open lazygit in a new tab:
`tab term lazygit`.
To automatically close the buffer when normal exitting with `q`, use an autocmd like this:
```lua
vim.api.nvim_create_autocmd("TermClose", {
  callback = function()
    if vim.v.event.status == 0 then
      vim.api.nvim_buf_delete(0, {})
    end
  end,
})

```

**Reply by asfaltboy** [reply]
This is almost perfect, but behaves weird if I hit `esc`, or maybe `esc esc`, the entire neovim stalls...

**Comment by zaynOm** [comment]
This is not working for me @dpetka2001 

```lua
map("n", "<leader>gg", function()
  Snacks.lazygit({ size = { width = 1, height = 1 }, cwd = LazyVim.root.get() })
end, { desc = "Lazygit (root dir)" })
```


Update: here the solution
```lua
{
   "folke/snacks.nvim",
   opts = {
     styles = {
       lazygit = {
         width = 0,
         height = 0,
       },
     },
   },
}
  ```

**Reply by dpetka2001** [reply]
It'd be better if you configure it in `snacks.nvim` now that it handles opening lazygit. Following should work
```lua
{
  "folke/snacks.nvim",
  opts = {
    styles = {
      lazygit = {
        height = 0,
        width = 0,
      },
    },
  }
}
```
You should also read the docs in `snacks.nvim` repo to know what other settings exist and how you can customize them.

**Reply by zaynOm** [reply]
Thx, i just found a solution by folke. i already checked `snacks` docs and wasn't able to find anything about this [here](https://github.com/folke/snacks.nvim/blob/main/docs/lazygit.md)

**Reply by dpetka2001** [reply]
Yes you didn't find anything because you only looked at lazygit's docs. But under the hood lazygit uses `Snacks.terminal` to open the terminal window with lazygit and that also uses `Snacks.win` to create the window for the terminal and that's where you'll find these options. So, they're all interchangeable and use each other and you would have to skim through all the parts to get the necessary info as a whole. Unfortunately, there's no other way. Just keep reading little by little and also ask questions about things unclear to you and someone will help you. That's what i try to do at least to the best of my ability and knowledge.
